vllm/model_executor/guided_decoding/outlines_logits_processors.py

```diff
@@ -77,8 +77,14 @@ def __call__(self, input_ids: List[int],
                 f"Unsupported instruction type {type(instruction)}")
 
         mask = torch.full((scores.shape[-1], ),
-                          -math.inf,
+                          -torch.inf,
                           device=scores.device)
+        # The tokenizer may support more token ids than the model can generate,
+        # eg. Llama 3.2 Vision models have an `<|image|>` token with id 128256
+        # but scores.shape == torch.Size([128256])
+        # Using NumPy is faster for filtering token ids
+        allowed_tokens = np.array(allowed_tokens)
+        allowed_tokens = allowed_tokens[allowed_tokens < scores.shape[-1]]
```

Comment:
@tjohnson31415 when you tried with pytorch did you copy to gpu first?
```python
allowed_tokens = torch.tensor(allowed_tokens, device=scores.device)
allowed_tokens = allowed_tokens.masked_select(allowed_tokens < scores.shape[-1])
mask.index_fill_(allowed_tokens, 0)
```

Reply:
Yep. Creating a torch.Tensor from a large list is oddly slow, whether on CPU side or GPU side it was about the same in my testing: 15ms for ~120k element list. Creating an `np.array` is less than 5ms. The rest of the operations are fast in either case 🤔

Reply:
Thanks @tjohnson31415! I was intrigued and tried this myself. I found the same thing as you, that creating a tensor from a python list was the most expensive part. But actually it's about 30% faster overall if you combine the two!
```python
allowed_tokens = np.array(allowed_tokens, dtype=np.int64)
allowed_tokens = torch.tensor(allowed_tokens, device=scores.device)
allowed_tokens = allowed_tokens.masked_select(allowed_tokens < scores.shape[-1])
mask.index_fill_(0, allowed_tokens, 0)
```

Reply:
I pushed an update with this change because @tjohnson31415 is OOO for a few days.